### PR TITLE
Triplelift Native: Mapping tag_code 

### PR DESCRIPTION
**Pull request by patrickloughrey**
Moving forward, Triplelift's native bid adapter has to copy the value of `imp.ext.data.tag_code` and set it as `imp.tagid` which Triplelift uses for placement mapping if the specified publisher is sending the request.

### CODE REVIEW

File: adapters/triplelift_native/triplelift_native.go [hunk 1]
```diff
@@ -53,7 +54,16 @@ func processImp(imp *openrtb2.Imp) error {
...
+	if req.Site != nil {
+		siteCopy := *req.Site
+		if siteCopy.Domain == "msn.com" {
+			imp.TagID = tlext.TagCode
+		} else {
+			imp.TagID = tlext.InvCode
+		}
+	} else {
+		imp.TagID = tlext.InvCode
+	}
```
Comment (gargcreation1992) [inline adapters/triplelift_native/triplelift_native.go:66]: Please add json test case for this change.

File: adapters/triplelift_native/triplelift_native.go [hunk 2]
```diff
@@ -89,7 +99,7 @@ func (a *TripleliftNativeAdapter) MakeRequests(request *openrtb2.BidRequest, ext
 	var validImps []openrtb2.Imp
 	// pre-process the imps
 	for _, imp := range tlRequest.Imp {
-		if err := processImp(&imp); err == nil {
+		if err := processImp(&imp, request); err == nil {
```
Comment (Kevin-P-Kerr) [inline adapters/triplelift_native/triplelift_native.go:102]: I think you need to pass in a pointer to the request (&request) and not the request itself

File: adapters/triplelift_native/triplelift_native.go [hunk 3]
```diff
@@ -53,7 +54,16 @@ func processImp(imp *openrtb2.Imp) error {
 	if tlext.InvCode == "" {
 		return fmt.Errorf("no inv_code specified")
 	}
-	imp.TagID = tlext.InvCode
+	if req.Site != nil {
+		siteCopy := *req.Site
```
Comment (Kevin-P-Kerr) [inline adapters/triplelift_native/triplelift_native.go:58]: do you need to declare siteCopy as a var somewhere?

**Review by Kevin-P-Kerr — CHANGES_REQUESTED** [review]

File: adapters/triplelift_native/triplelift_native.go (hunk 2 above)
Comment (patrickloughrey) [reply adapters/triplelift_native/triplelift_native.go:102]: I believe this is ok. We already passing a pointer (via the MakeRequests arg)

**Review by Sonali-More-Xandr — APPROVED** [review]

File: adapters/triplelift_native/triplelift_native.go [hunk 4]
```diff
@@ -53,7 +69,21 @@ func processImp(imp *openrtb2.Imp) error {
...
+		extData = *ext.Data
+	}
+
+	if extData.TagCode != "" {
+		if siteCopy.Publisher.Domain == "msn.com" {
+			fmt.Println(extData.TagCode)
+			imp.TagID = extData.TagCode
+		} else {
+			imp.TagID = tlext.InvCode
+		}
```
Comment (bsardo) [inline adapters/triplelift_native/triplelift_native.go:83]: This conditional is missing test coverage. You might want to consider adding a JSON test to cover this case.

Comment (patrickloughrey) [reply adapters/triplelift_native/triplelift_native.go:82]: optional-params.json accounts for this case

Comment (bsardo) [reply adapters/triplelift_native/triplelift_native.go:82]: This `else` does not appear to be covered. The outer `else` with the same assignment just after this block is covered though.

**Review by bsardo — APPROVED** [review]

**Review by przemkaczmarek — APPROVED** [review]

**Pull request merged**